ktor-server/ktor-server-plugins/ktor-server-sessions/jvm/src/io/ktor/server/sessions/SessionStorage.kt

```diff
@@ -28,14 +23,9 @@ public interface SessionStorage {
     public suspend fun invalidate(id: String)
 
     /**
-     * Reads session [id] using a [consumer] as [R]
-     *
-     * This method calls the [consumer] with a [ByteReadChannel] and it is in charge of the channel's lifecycle,
-     * and returns the object [R] produced by the [consumer].
-     * [consumer] should read the content of the specified [ByteReadChannel] and return an object of type [R].
+     * Reads session by [id]
      *
-     * @return instance of [R] representing the session object returned from the [consumer].
      * @throws NoSuchElementException when session [id] is not found.
      */
-    public suspend fun <R> read(id: String, consumer: suspend (ByteReadChannel) -> R): R
+    public suspend fun read(id: String): String
```

Comment:
Same, this will break existing user implementations :(

Reply:
that's ok, we're going to make it in `2.0.0` with other breaking changes.

@rsinukov please make sure we have migrations updated

Reply:
Same here, ByteArray in API looks better